src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
```

Comment:
You can simplify `else` statement if you immediately add to prev field in constructor tail-node
```suggestion
        Node<T> node = new Node<>(tail, value, null);
```